src/tests/ftest/util/network_utils.py

```diff
@@ -0,0 +1,560 @@
+
+    """
+    log = getLogger()
+
+    # Find any active network interfaces on the server or client hosts
+    interfaces_per_host = get_active_network_interfaces(hosts, verbose)
+
+    # From the active interface dictionary find all the interfaces that are common to all hosts
+    log.info("Active network interfaces detected:")
+    common_interfaces = []
+    for interface, node_set in interfaces_per_host.items():
+        log.info("  - %-8s on %s (Common=%s)", interface, node_set, node_set == hosts)
+        if node_set == hosts:
+            common_interfaces.append(interface)
+
+    return common_interfaces
+
+
+def get_interface_speeds(hosts, interface, verbose=True):
+    """Get the speeds of this network interface on each host.
+
+    Args:
+        hosts (NodeSet): hosts on which to detect the interface speed
+        interface (str): interface for which to obtain the speed
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        dict: a dictionary of interface speed keys and NodeSet values on which they were detected
+
+    """
+    net_path = os.path.join(os.path.sep, "sys", "class", "net")
+    command = "cat {}".format(os.path.join(net_path, interface, "speed"))
+    task = run_task(hosts, command, verbose=verbose)
+    if verbose:
+        display_task(task)
+
+    # Populate a dictionary of interface speeds with a NodSet of hosts on which it was detected
+    interface_speeds = {}
+    results = dict(task.iter_retcodes())
+    if 0 in results:
+        for output, nodelist in task.iter_buffers(results[0]):
+            output_lines = [line.decode("utf-8") for line in output]
+            nodeset = NodeSet.fromlist(nodelist)
+            for line in output_lines:
+                try:
+                    speed = int(line.strip())
+                except IOError as io_error:
+                    # KVM/Qemu/libvirt returns an EINVAL
+                    if io_error.errno == errno.EINVAL:
+                        speed = 1000
+                except ValueError:
+                    # Any line not containing a speed (integer)
+                    continue
+                if speed not in interface_speeds:
+                    interface_speeds[speed] = NodeSet()
+                interface_speeds[speed].update(nodeset)
+
+    return interface_speeds
+
+
+def get_interface_numa_node(hosts, interface, verbose=True):
+    """Get the NUMA node ID of this network interface on each host.
+
+    Args:
+        hosts (NodeSet): hosts on which to detect the NUMA node
+        interface (str): interface for which to obtain the NUMA node
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        dict: a dictionary of NUMA node ID keys and NodeSet values on which they were detected
+
+    """
+    net_path = os.path.join(os.path.sep, "sys", "class", "net")
+    command = "cat {}".format(os.path.join(net_path, interface, "device", "numa_node"))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string